theseus/theseus_layer.py

```diff
@@ -93,3 +103,88 @@ def device(self) -> torch.device:
     @property
     def dtype(self) -> torch.dtype:
         return self.objective.dtype
+
+
+def _forward(objective, optimizer, optimizer_kwargs, input_data):
+    objective.update(input_data)
+    info = optimizer.optimize(**optimizer_kwargs)
+    vars = [var.data for var in objective.optim_vars.values()]
+    return vars, info
+
+
+class TheseusLayerDLMForward(torch.autograd.Function):
+    """
+    Functionally the same as the forward method in a TheseusLayer
+    but computes the direct loss minimization in the backward pass.
+    """
+
+    @staticmethod
+    def forward(ctx, objective, optimizer, optimizer_kwargs, input_data, epsilon, *params):
+        optim_vars, info = _forward(objective, optimizer, optimizer_kwargs, input_data)
```

Comment:
nit: someone used to Theseus terminology might see `optim_vars` and think these are `th.Variable` objects. How about `optim_tensors`? Similar comment for usage in `backward`.

Reply:
Good idea.